### ISSUE TITLE
Can't move groups

### ISSUE THREAD

**Issue by MrDOS** [eduard-permyakov/permafrost-engine#3]
In the demo, if I select more than one unit and attempt to move them by right-clicking, the process dies with an assertion failure:

```
pf: src/navigation//nav.c:1000: N_DesiredVelocity: Assertion `ret == id' failed.
```

I have no problems moving single units, and I think this problem occurs with any two or more units regardless of type.

**Comment by eduard-permyakov** [comment]
This but is not reproducible by me unfortunately. I am able to move groups of units freely without ever encountering this failed assert. Perhaps there are more specific preconditions that must be met. Or this may be a symptom of a more subtle bug caused by different build/hardware configurations.

**Comment by rtri** [comment]
Hi,

I also ran into this assert while checking out your flowfield code. The problem is that `dest_id_t id` in `movement.c::make_flock_from_selection` is not initialized if the left-hand side of `if(adjacent_to_any_in_set(curr_ent, pathed_ents, num_pathed_ents)
        || M_NavRequestPath(s_map, (vec2_t){curr_ent->pos.x, curr_ent->pos.z}, target_xz, &id))` evaluates to true, but will still overwrite the flock's destination id via `new_flock.dest_id = id` even when a previous iteration already assigned it. A simple fix is to replace

        dest_id_t id;
        if(adjacent_to_any_in_set(curr_ent, pathed_ents, num_pathed_ents)
        || M_NavRequestPath(s_map, (vec2_t){curr_ent->pos.x, curr_ent->pos.z}, target_xz, &id))

by

        if(adjacent_to_any_in_set(curr_ent, pathed_ents, num_pathed_ents)
        || M_NavRequestPath(s_map, (vec2_t){curr_ent->pos.x, curr_ent->pos.z}, target_xz, &new_flock.dest_id))

though you may of course know of a better one.

nice shades of Carmack btw :)

**Comment by eduard-permyakov** [comment]
Thanks a lot for the detailed investigation @rtri ! And thanks for the compliment :blush:. I have applied the exact fix you have suggested. The fix is in commit e06757dfe333f0f6778ec7dadf2716ad3c6460fe.
